gramps/plugins/db/dbapi/dbapi.py

```diff
@@ -83,6 +82,21 @@ class DBAPI(DbGeneric):
     def _initialize(self, directory, username, password):
         raise NotImplementedError
 
+    def supports_json_access(self):
+        """
+        A DBAPI level method for testing if the
+        database support JSON access.
+        """
+        try:
+            self.dbapi.execute("SELECT json_data from person limit 1;")
+            self.dbapi.fetchone()
+            self.dbapi.execute("SELECT json_data from metadata limit 1;")
+            self.dbapi.fetchone()
+            self.dbapi.commit()
+            return True
+        except Exception:
```

Comment:
What if an exception is thrown for a reason other than the json_data column does not exist in the table?
In this scenario, you do not know if the DB support JSON or not. Maybe it is best to let the exception go upwards in this case - there is probably nothing you can do as you cannot determine if the DB is using blob or json.

Reply:
That is possible, but defeats the purpose of the function to return True or False. There are better ways to check to see if a column exists in specific implementations of SQL (such as using PRAGMA in SQLite), but as far as I know, there is not a DB-API level check. Overrides can do something more specific.

Reply:
Sorry, re-reading my comment I realise it's not very clear. Here's some pseudo-python in which the 3 scenarios are hopefully clearer

```
try:
  self.dbapi.execute("SELECT json_data from metadata limit 1;")
  self.dbapi.fetchone()
  self.dbapi.commit()
  return True
except DbException: # this may not be the correct type of exception to catch
  # handle *only* an exception raised due to the column json_data not existing
  return False
except:
  #  for all other exceptions, re-throw since we don't know if True of False is the correct answer
  raise
```

Reply:
I see. Yes that makes sense. Thanks!

Reply:
The error is a `OperationalError`. However, that is a SQLite error. 

I could require DB-API backends (like SQLite) to implement this method. 

What do you think?

Reply:
We don't want SQLite implementation detail to leak into this layer. 
If we step back, the question is "is the DB we are opening JSON or blob (or conceptually something else in future)"
The current approach is to select up to 1 row from metadata and infer the answer from a lack of error.
What if you change the approach and ask the DB layer if the column 'json_data' exists in the table metadata? This feels more generic and is more directly the question we want to answer. The db layer can handle the True/False/Exception based on DB specific knowledge. The downside is that it extends the db layer and I assume don't immediately have another use for it.

Reply:
I cleaned up this code, and I think it is much better. Everything is at the correct level. It does require DB-API databases to add one new method, `column_exists()` which goes with `table_exists()` that was already present.

Reply:
I think this is resolved. If not, let me know :)